hazelcast/src/main/java/com/hazelcast/collection/impl/collection/CollectionService.java

```diff
@@ -90,8 +88,12 @@ public void shutdown(boolean terminate) {
 
     @Override
     public void destroyDistributedObject(String name) {
-        getContainerMap().remove(name);
-        nodeEngine.getEventService().deregisterAllListeners(getServiceName(), name);
+        CollectionContainer container = getContainerMap().remove(name);
+        if (container != null) {
+            container.clear(false);
+            container.clearBackup();
+            nodeEngine.getEventService().deregisterAllListeners(getServiceName(), name);
```

Comment:
Not sure, but I think you might still this outside the IF.
Considering that the data-structure existed in the second brain, that got merged into the first brain (where it didn't exist), you still want to release these resource on the member from the second brain. No?